libethcore/SealEngine.h

```diff
@@ -138,7 +138,13 @@ class NoProof: public eth::SealEngineBase
     static std::string name() { return "NoProof"; }
     static void init();
     void generateSeal(BlockHeader const& _bi) override;
+    void populateFromParent(BlockHeader& _bi, BlockHeader const& _parent) const override;
 };
 
+u256 calculateDifficulty(
```

Comment:
Call it `calculateEthashDifficulty`

Reply:
Sounds weird calling EthashDifficulty from NoProof

Reply:
right, but that's what we need to do, I think it better expresses what's happening

Reply:
is it also calculateEthashGasPrice ?

Reply:
I'd leave `calculateGasLimit`, it doesn't seem to me to be inherent to Ethash
(could also go to the base class, but this global function is fine, too)